### PR TITLE
Adds test case where use up all unreserved registers.

### PR DESCRIPTION
**Pull request by mathetake**
Signed-off-by: Takeshi Yoneda <takeshi@tetrate.io>

### CODE REVIEW

**Comment by mathetake** [comment]
intends to resolve #328, but in anyway this helps us to find bugs like #311

File: tests/engine/testdata/hugestack.wat [hunk 1]
```diff
@@ -0,0 +1,24 @@
+(module
```
Comment (mathetake) [inline tests/engine/testdata/hugestack.wat:1]: I think we need formatter for wat :D

**Comment by mathetake** [comment]
so found that no need to save callee saved registers in arm64 ABI, but we are on Go's ABI so it's only necessary not to use X18 and X28

**Comment by mathetake** [comment]
for arm64

File: tests/engine/adhoc_test.go [hunk 2]
```diff
@@ -67,28 +72,50 @@ func runTests(t *testing.T, newRuntimeConfig func() *wazero.RuntimeConfig) {
 	})
 }
 
-func testFibonacci(t *testing.T, newRuntimeConfig func() *wazero.RuntimeConfig) {
+func testHugeStack(t *testing.T, newRuntimeConfig func() *wazero.RuntimeConfig) {
 	// We execute 1000 times in order to ensure the JIT engine is stable under high concurrency
```
Comment (mathetake) [inline tests/engine/adhoc_test.go:76]: I think in another PR I will separate out the counccurent testing in conccurent_test.go

**Review by codefromthecrypt — APPROVED** [review]
good find!

**Pull request merged**